src/components/FilterAttachedResults/FilterAttachedResults.ts

```diff
@@ -0,0 +1,149 @@
+import {
+    Component,
+    ComponentOptions,
+    IComponentBindings,
+    $$,
+    Initialization,
+    Checkbox,
+    QueryEvents,
+    IBuildingQueryEventArgs,
+    QueryStateModel,
+    load,
+    IAttributesChangedEventArg
+} from 'coveo-search-ui';
+
+export interface IAnalyticsAttachedResultsFilterMeta {
+    filteredAttachedResults: boolean;
+}
+
+export interface IFilterAttachedResultsOptions {
+    /** Specifies the text displayed next to the checkbox. */
+    text?: string;
+    /** The function that is called when retrieving uri hashes to filter. */
+    getUriHashes?: () => string[];
+}
+
+/**
+ * The FilterAttachedResults component allows a user to click a checkbox to
+ * search only for attached results.
+ */
+export class FilterAttachedResults extends Component {
+    static ID = 'FilterAttachedResults';
+
+    private checkbox: Checkbox;
+
+    static options: IFilterAttachedResultsOptions = {
+        text: ComponentOptions.buildStringOption({
+            defaultValue: 'Filter Attached Results'
+        }),
+        getUriHashes: ComponentOptions.buildCustomOption(
+            name => () => new Array<string>(),
+            { defaultFunction: () => () => new Array<string>() }
+        )
+    };
+
+    constructor(
+        public element: HTMLElement,
+        public options: IFilterAttachedResultsOptions,
+        public bindings?: IComponentBindings
+    ) {
+        super(element, FilterAttachedResults.ID, bindings);
+
+        this.options = ComponentOptions.initComponentOptions(
+            element,
+            FilterAttachedResults,
+            options
+        );
+
+        this.initialize();
+
+        this.queryStateModel.registerNewAttribute(
+            QueryStateModel.getFacetId(FilterAttachedResults.ID),
+            false
+        );
+        this.bind.onRootElement(
+            QueryEvents.buildingQuery,
+            (args: IBuildingQueryEventArgs) => this.handleBuildingQuery(args)
+        );
+        this.bind.onQueryState(
+            'change:',
+            QueryStateModel.getFacetId(FilterAttachedResults.ID),
+            (args: IAttributesChangedEventArg) =>
+                this.handleQueryStateChange(args)
```

Comment:
this.handleQueryStateChange.bind(this) instead of an arrow ?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind